tests/forms_tests/field_tests/test_decimalfield.py

```diff
@@ -54,13 +42,25 @@ def test_decimalfield_1(self):
             f.clean('-000.123')
         with self.assertRaisesMessage(ValidationError, "'Ensure that there are no more than 4 digits in total.'"):
             f.clean('-000.12345')
-        with self.assertRaisesMessage(ValidationError, "'Enter a number.'"):
-            f.clean('--0.12')
         self.assertEqual(f.max_digits, 4)
         self.assertEqual(f.decimal_places, 2)
         self.assertIsNone(f.max_value)
         self.assertIsNone(f.min_value)
 
+    def test_enter_a_number_error(self):
+        f = DecimalField(max_digits=4, decimal_places=2)
+        values = (
+            '-NaN', 'NaN', '+NaN',
+            '-sNaN', 'sNaN', '+sNaN',
+            '-Inf', 'Inf', '+Inf',
+            '-Infinity', 'Infinity', '+Infinity',
+            'a', 'łąść', '1.0a', '--0.12',
+        )
+        for value in values:
+            with self.subTest(value=value):
+                with self.assertRaisesMessage(ValidationError, "'Enter a number.'"):
```

Comment:
Both `with` should fit on the same line,

`with self.subTest(value=value), self.assertRaisesMessage(ValidationError, "'Enter a number.'"):`